src/listContext.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+import getToken from './lib/token';
+import firebase from 'firebase/app';
+
+const ListContext = React.createContext();
+
+const ListContextProvider = props => {
+  const initialToken = () => window.localStorage.getItem('token') || getToken();
+
+  const [token] = useState(initialToken);
+  const [shoppingList, setShoppingList] = useState([]);
+
+  // getting the token from localStorage
+  useEffect(() => {
+    window.localStorage.setItem('token', token);
+  }, [token]);
+
+  const fetchList = token => {
```

Comment:
Created this function to fetch the list from the database if the user enters directly to the addItem page